File: native/n2_cmake/src/main/java/com/mic/jnibase/MainActivity.java
package com.mic.jnibase;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    static {
        // 动态库情况下：加载动态库到内存空间
        System.loadLibrary("getndk");
        System.loadLibrary("native-lib");
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        TextView tv = findViewById(R.id.sample_text);
        tv.setText(stringFromJNI());
    }

    public native String stringFromJNI();
}

File: dagger/src/main/java/com/mic/MainActivity.kt
package com.mic

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.mic.jetpack.dagger2.Dagger2Fragment


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        this.supportFragmentManager.beginTransaction().add(R.id.root,Dagger2Fragment()).commit()
    }
}
